src/components/data-management/ProjectDetails.jsx

```diff
@@ -24,6 +29,47 @@ const ProjectDetails = ({ width, height }) => {
 
   const [data, setData] = useState([]);
   const [sortedSpeciesData, setSortedSpeciesData] = useState([]);
+  const projects = useSelector((state) => state.projects);
+  const samples = useSelector((state) => state.samples);
+
+  const uploadFiles = (filesList, sampleType) => {
+    const samplesMap = filesList.reduce((acc, file) => {
+      const sampleName = file.name.trim().replace(/[\s]{2,}/ig, ' ').split('/')[0];
+      const sampleUuid = Object.values(samples).filter(
+        (s) => s.name === sampleName
+          && s.projectUuid === projects.meta.activeProject,
+      )[0]?.uuid;
+
+      return {
+        ...acc,
+        [sampleName]: {
+          ...acc[sampleName],
+          uuid: sampleUuid,
+          files: {
+            ...acc[sampleName]?.files,
+            [sampleName]: file,
+          },
+        },
+      };
+    }, {});
```

Comment:
Mapped first according to sample name so that for each sample, only one `dispatch` needs to be called to set up a uuid. If the loop is run for each file, each file will call a dispatch to create sample, which will not group the samples together.